### ISSUE TITLE
Use `higher_is_better` from `torch_metrics` instead of manual `MIN` and `MAX`

### ISSUE THREAD

**Issue by yinweisu** [awslabs/autogluon#1799]
@gradientsky pointed out it might be better to use `higher_is_better` from [torchmetrics](https://torchmetrics.readthedocs.io/en/stable/pages/overview.html#metrics-and-hyperparameter-optimization)

@sxjscience @zhiqiangdon @bryanyzhu we can discuss here to share opinions.

**Comment by zhiqiangdon** [comment]
Using `higher_is_better ` is good. There are maybe some special cases to consider. For example, we use torchmetrics.MeanMetrc + pytorch's cross entropy function to compute the `log_loss` or `cross_entropy` metric. In these cases, we may need to manually set the `higher_is_better` attribute in torchmetrics.MeanMetrc's objects.
